### DISCUSSION TITLE
Installation help?

### DISCUSSION THREAD

**Discussion by justinchiao** [JosephLai241/URS#62] [Q&A]
git clone --depth=1 https://github.com/JosephLai241/URS.git
cd URS

I can't get past here. 
poetry install
poetry shell
maturin develop --release

I am pretty sure i have poetry already installed. When I try to run these last three lines, i get 'poetry' is not recognized as an internal or external command, operable program or batch file.

Thank you

**Comment by JosephLai241** [comment]
> I am pretty sure i have poetry already installed.

Make sure you have it installed by using one of the installation methods as shown in the [Installation section of the documentation](https://python-poetry.org/docs/#installation). I recommend installing via the official installer.

If you still cannot invoke `Poetry`, this is probably a `$PATH` issue. You need to add `Poetry` to your `$PATH` as shown in step 3 on the installation process. This is ripped from that section of the documentation:

> The installer creates a `poetry` wrapper in a well-known, platform-specific directory:
> 
> - `$HOME/.local/bin` on Unix.
> - `%APPDATA%\Python\Scripts` on Windows.
> - `$POETRY_HOME/bin` if `$POETRY_HOME` is set.
> 
> If this directory is not present in your `$PATH`, you can add it in order to invoke Poetry as `poetry`.
> 
> Alternatively, the full path to the `poetry` binary can always be used:
> 
> - `~/Library/Application Support/pypoetry/venv/bin/poetry` on MacOS.
> - `~/.local/share/pypoetry/venv/bin/poetry` on Linux/Unix.
> - `%APPDATA%\pypoetry\venv\Scripts\poetry` on Windows.
> - `$POETRY_HOME/venv/bin/poetry` if `$POETRY_HOME` is set.

[This article](https://janelbrandon.medium.com/understanding-the-path-variable-6eae0936e976) may be helpful for getting a basic understanding of what `$PATH` is.
